ttm_module.gradle

```diff
@@ -0,0 +1,43 @@
+buildscript {
+    repositories {
+        maven {
+            url 'https://artifactory.trimble.tools/artifactory/ttm-mobile-ecosystem-maven'
+            credentials {
+                username = "${artifactory_user}"
+                password = "${artifactory_password}"
+            }
+        }
+    }
+}
+
+artifactory {
+    contextUrl = "${artifactory_contextUrl}"   //The base Artifactory URL if not overridden by the publisher/resolver
+    resolve {
+        repository {
+            repoKey = 'ttm-mobile-ecosystem-maven'
+            username = "${artifactory_user}"
+            password = "${artifactory_password}"
+            maven = true
+
+        }
+    }
+}
+
+android {
+    defaultConfig {
+        archivesBaseName = "${project.getName()}-$versionName"
+    }
+    signingConfigs {
+        release {
+            storeFile project.findProperty("keystore_path") ? rootProject.file(property("keystore_path")) : null
+            storePassword project.findProperty("keystore_password")
+            keyAlias project.findProperty("key_alias")
+            keyPassword project.findProperty("key_password")
+        }
+    }
+    buildTypes {
```

Comment:
You'd also need:
buildscript {
    dependencies {
        classpath "org.jetbrains.dokka:dokka-android-gradle-plugin:0.9.17"
    }
}

Reply:
I looked into it, and I'm not able to do that because of the Gradle limitation that prevents plugins from being applied from script plugins. Basically, the plugin has to be declared after the dependency but before dokka, and it can't be in the script plugin, so our options are to either create another script plugin just for setting up the dependencies or to add those dependencies to the instructions in the README. Either way, a new plugin would need to be added to those instructions.

Reply:
Actually, I can't create a separate script to add the dependency. It doesn't work properly, so the plugin still doesn't resolve. This is true whether I put it in the project or the module itself. It seems like, until Gradle removes the restriction on applying plugins in script plugins, we will have to manage Dokka on a project-by-project basis.